pkg/kapp/resources/resources.go

```diff
@@ -100,7 +100,7 @@ func (c *ResourcesImpl) All(resTypes []ResourceType, opts AllOpts) ([]Resource,
 	c.opts.FallbackAllowedNamespaces = uniqAndValidNamespaces(append(c.opts.FallbackAllowedNamespaces, opts.ResourceNamespaces...))
 
 	nsScope := "" // all namespaces by default
-	nsScopeLimited := c.opts.ScopeToFallbackAllowedNamespaces && len(c.opts.FallbackAllowedNamespaces) == 1
+	nsScopeLimited := c.opts.ScopeToFallbackAllowedNamespaces && len(c.opts.FallbackAllowedNamespaces) >= 1
```

Comment:
So we went with `>=1`, but wouldn't the value `o.NamespaceFlag.Name` will always be included. So the second condition would always be true?

Reply:
And even though the condition [here](https://github.com/vmware-tanzu/carvel-kapp/pull/533/commits/3ce429f925682f2c8effe264be6fe2205ae0fc28#diff-4bf6c9b48383fc3b53344634e9527785c014d0aa647140a49c63d76bc1b22918L107-L110) is never met.

I think it should now be
```
if c.opts.ScopeToFallbackAllowedNamespaces && len(c.FallbackAllowedNamespaces) == 1
```
(Case where it makes a single list call)